tests/test_hierarchy.cc

```diff
@@ -220,3 +223,118 @@ BOOST_AUTO_TEST_CASE(zoltan)
     BOOST_TEST(conv_rate == ref_solution, tt::tolerance(1e-6));
   }
 }
+
+dealii::TrilinosWrappers::SparseMatrix
+gimme_a_matrix(unsigned int n_local_rows, unsigned int n_entries_per_row)
+{
+  unsigned int const comm_size =
+      dealii::Utilities::MPI::n_mpi_processes(MPI_COMM_WORLD);
+  unsigned int const comm_rank =
+      dealii::Utilities::MPI::this_mpi_process(MPI_COMM_WORLD);
+  unsigned int const size = comm_size * n_local_rows;
```

Comment:
This only works if `n_local_rows` is the same on every processor, which is fine, but add a comment to make it clear.